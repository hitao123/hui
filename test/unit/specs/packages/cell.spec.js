import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Cell from '../../../../packages/cell';
import CellDemo from '../../../../docs/src/views/cell';

describe('Cell component', () => {
  it('should show Cell', () => {
    const warpper = shallowMount(Cell, {
      propsData: {
        title: 'Cell'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show CellDemo', () => {
    expect(renderToString(CellDemo)).toMatchSnapshot();
  });
});
