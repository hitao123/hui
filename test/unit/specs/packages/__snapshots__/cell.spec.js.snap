// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component should show Cell 1`] = `
<div class="h-cell h-hairline">
  <!---->
  <div class="h-cell__title"><span>Cell</span>
    <div class="h-cell__label"></div>
  </div>
  <div class="h-cell__value"><span></span></div>
  <!---->
</div>
`;

exports[`Cell component should show CellDemo 1`] = `
<div>
  <div class="demo-cell">
    <h-cell-group>
      <h-cell title="单元格"></h-cell>
      <h-cell title="单元格" value="内容"></h-cell>
    </h-cell-group>
  </div>
  <div class="demo-cell">
    <h-cell-group>
      <h-cell title="单元格" value="内容" label="描述"></h-cell>
    </h-cell-group>
  </div>
  <div class="demo-cell">
    <h-cell-group>
      <h-cell value="内容" label="描述"></h-cell>
    </h-cell-group>
  </div>
  <div class="demo-cell">
    <h-cell-group>
      <h-cell icon="wechat" value="内容" label="描述"></h-cell>
    </h-cell-group>
  </div>
  <div class="demo-cell">
    <h-cell-group>
      <h-cell title="单元格" value="内容" isLink=""></h-cell>
      <h-cell title="单元格" value="内容" isLink="" arrow-direction="down"></h-cell>
    </h-cell-group>
  </div>
</div>
`;
